selfdrive/boardd/panda.cc

```diff
@@ -35,10 +29,11 @@ Panda::Panda(std::string serial) {
       if (dev_handle == NULL) { goto fail; }
 
       unsigned char desc_serial[25];
-      int ret = libusb_get_string_descriptor_ascii(dev_handle, desc.iSerialNumber, desc_serial, sizeof(desc_serial));
+      int ret = libusb_get_string_descriptor_ascii(dev_handle, desc.iSerialNumber, desc_serial, std::size(desc_serial));
       if (ret < 0) { goto fail; }
 
-      if (serial.empty() || serial.compare(reinterpret_cast<const char*>(desc_serial)) == 0) {
+      usb_serial = std::string(desc_serial, desc_serial+std::size(desc_serial)).c_str();
```

Comment:
```suggestion
      usb_serial = std::string(desc_serial, ret);
```

Reply:
Same as above, that's what I tried first.